orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestForceCacheRefreshTask.java

```diff
@@ -62,24 +66,41 @@
   private final long timeout = TimeUnit.MINUTES.toMillis(15);
 
   private final long autoSucceedAfterMs = TimeUnit.MINUTES.toMillis(12);
+  private final Clock clock;
+  private final Registry registry;
+  private final CloudDriverCacheService cacheService;
+  private final CloudDriverCacheStatusService cacheStatusService;
+  private final ObjectMapper objectMapper;
 
   @Autowired
-  CloudDriverCacheService cacheService;
-
-  @Autowired
-  CloudDriverCacheStatusService cacheStatusService;
+  public ManifestForceCacheRefreshTask(Registry registry,
+                                       CloudDriverCacheService cacheService,
+                                       CloudDriverCacheStatusService cacheStatusService,
+                                       ObjectMapper objectMapper) {
+    this(registry, cacheService, cacheStatusService, objectMapper, Clock.systemUTC());
+  }
 
-  @Autowired
-  ObjectMapper objectMapper;
+  ManifestForceCacheRefreshTask(Registry registry,
+                                CloudDriverCacheService cacheService,
+                                CloudDriverCacheStatusService cacheStatusService,
+                                ObjectMapper objectMapper, Clock clock) {
+    this.registry = registry;
+    this.cacheService = cacheService;
+    this.cacheStatusService = cacheStatusService;
+    this.objectMapper = objectMapper;
+    this.clock = clock;
+  }
 
   @Override
   public TaskResult execute(Stage stage) {
     Long startTime = stage.getStartTime();
     if (startTime == null) {
       throw new IllegalStateException("Stage has no start time, cannot be executing.");
     }
-    if ((System.currentTimeMillis() - startTime) > autoSucceedAfterMs) {
+    if ((clock.millis() - startTime) > autoSucceedAfterMs) {
       log.info("{}: Force cache refresh never finished processing... assuming the cache is in sync and continuing...", stage.getExecution().getId());
+      registry.counter(String.format("%s.timeoutSuccess", METRIC_PREFIX)).increment();
```

Comment:
I think it would be nice to have a single timer with a tag to distinguish the outcomes.
The test at #100 looks suspicious (even though it is analogous to before). I'm not familiar with the code but suggests that autoSucceed is there in case getStartTime took a long time. I assume that makes sense.
If you had a common timer, you would be able to know how much time we are talking about for these cases, plus would be able to relate that to overall attempts.

Also in the final else clause (incidentally the "else" is superfluous since the if already returned) you have no metric. So if you want to know how long it takes refesh to succeed, and the failures are still going through that refreshManifests call, I'd think you'd want to update that timer with that else case as well. Besides giving complete coverage over the refreshManifests time, it would tell you how often you are going through each of the code paths.

Perhaps the outcome could have string values indicating timeout success, refresh success, and whatever that third case is.

If it makes sense, you coudl add a "success" tag and also include that initial exception. If getStartTime is so complicated that you are worrying about timeouts, then you might also be interested in failures.

Reply:
@robzienert has more specific expertise about instrumenting within orca. He might have additional comments.

Reply:
Yeah I think a single timer for success and failure will be good. I'd like to leave a separate counter specifically for this case though, when the 12 minute time limit for the Force Cache Refresh is hit.

Reply:
wouldnt that case be the same as "statistic=count" and "outcome=AUTO_SUCCESS" or whatever you called the outcome?